src/main/java/com/r307/arbitrader/service/TradingService.java

```diff
@@ -146,8 +146,12 @@ public void setActivePosition(ActivePosition activePosition) {
     private void enterPosition(Spread spread) {
         final String longExchangeName = spread.getLongExchange().getExchangeSpecification().getExchangeName();
         final String shortExchangeName = spread.getShortExchange().getExchangeSpecification().getExchangeName();
-        final BigDecimal longFeePercent = exchangeService.getExchangeFee(spread.getLongExchange(), spread.getCurrencyPair(), true);
-        final BigDecimal shortFeePercent = exchangeService.getExchangeFee(spread.getShortExchange(), spread.getCurrencyPair(), true);
+        final BigDecimal tradeFee = exchangeService.getExchangeFee(spread.getLongExchange(), spread.getCurrencyPair(), true).getTradeFee();
+        // When we short trade a coin, the fee is always: marginFee + tradeFee
+        final ExchangeFee shortExchangeFee = exchangeService.getExchangeFee(spread.getShortExchange(), spread.getCurrencyPair(), true);
+        final BigDecimal marginFee = shortExchangeFee.getMarginFee()
+            .orElseThrow(() -> new RuntimeException("Missing required margin fee for exchange " + longExchangeName))
+            .add(shortExchangeFee.getTradeFee());
```

Comment:
The variable names confused me:
ExchangeFee has two fees: tradeFee and marginFee
But here:
tradeFee is the long exchange tradeFee
marginFee is the sum of tradeFee and marginFee for the longExchange

To avoid confusion we could use different variable names:
longTradeFee (= tradeFee)
shortTotalFee (= tradeFee + marginFee)

Reply:
Agreed. Once you take the `marginFee` and add the `tradeFee` to it, it is specific to the short trade and it is the total of both the fees that apply to that trade.